test/trivial/deitz/coerce-assign/plusassign.chpl

```diff
@@ -5,6 +5,7 @@ class D {
 class C : D {
   var x = 1;
   proc foo() {
+    pragma "last resort"
```

Comment:
What makes this necessary?  Ohhh...  the static type being `C` makes it prefer a generic overload over one taking a D?  That gives me slight pause...

Reply:
If the formal were simply `D` rather than `borrowed D` would that change things (since both would be generic?).  Is there an equivalent to this pattern in C++ and would it make the same choice?

Reply:
> If the formal were simply `D` rather than `borrowed D` would that change things (since both would be generic?).

No, because the change in behavior here just has to do with the `proc bar(_x)` being preferred or not preferred vs. one with an implicit conversion.

Here is a more focused Chapel program showing the behavior:

``` chapel
class Parent { }

class Child : Parent { }

proc bar(x) {
  writeln("bar(x)");
}
proc bar(x: borrowed Parent) {
  writeln("bar(x: borrowed Parent)");
}

var c = (new owned Child()).borrow();
bar(c);
// before PR: bar(x: borrowed Parent)
// after  PR: bar(x)

proc baz(x) {
  writeln("baz(x)");
}
proc baz(x: Parent) {
  writeln("baz(x: Parent)");
}
baz(c);
// before PR: baz(x: Parent)
// after  PR: baz(x)
```

Here is a similar C++ program:

``` c++
#include <cstdio>

struct Parent { int unused1; };

struct Child : Parent { int unused2; };

template<typename T>
void bar(T x) {
  printf("bar(T x)\n");
}

void bar(const Parent& x) {
  printf("bar(const Parent& x)\n");
}

int main() {
  Child c;
  bar(c); // prints bar(T x)
}
```

I don't think this is particularly different from a scenario like `proc foo(arg: int)` vs `proc foo(arg)` when passed `myInt8`.